### PR TITLE
btcutil: format BTC amounts with trailing zeroes

### LINKED ISSUE #1995: BTC amounts show trailing zeroes
When showing smaller satoshi amounts as full BTC, for example in lnd (`tx_fee=0.0000325 BTC`), it is hard to understand the actual amount used at first glance. Is this 325sat? 3250sat? 32500sat?

By adding trailing zeroes to this amount, it is easier to reason about the actual satoshis used:

`tx_fee=0.00003250 BTC` clearly means 3250sat.

### PR DESCRIPTION
**Pull request by C-Otto**
so that sat amounts can be read without counting zeroes

before:
```
 350sat = 0.0000035 BTC
3500sat = 0.000035 BTC
```

after:
```
 350sat = 0.00000350 BTC
3500sat = 0.00003500 BTC
```

fixes #1995

### CODE REVIEW

File: btcutil/amount.go [hunk 1]
```diff
@@ -101,11 +102,23 @@ func (a Amount) ToBTC() float64 {
...
 // string for a given unit.  The conversion will succeed for any unit,
-// however, known units will be formated with an appended label describing
+// however, known units will be formatted with an appended label describing
 // the units with SI notation, or "Satoshi" for the base unit.
 func (a Amount) Format(u AmountUnit) string {
 	units := " " + u.String()
-	return strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64) + units
+	formatted := strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64)
+
+	// when formatting full BTC, add trailing zeroes for numbers with decimal
```
Comment (guggero) [inline btcutil/amount.go:111]: nit: missing capitalization and punctuation in sentence.

File: btcutil/amount.go [hunk 2]
```diff
@@ -101,11 +102,23 @@ func (a Amount) ToBTC() float64 {
...
+	// point to ease reading of sat amount
+	if u == AmountBTC {
+		decimalPoint := "."
+		before, after, containsDecimalPoint := strings.Cut(formatted, decimalPoint)
+		if containsDecimalPoint {
+			trailingZeroes := strings.Repeat("0", 8-len(after))
+			return before + decimalPoint + after + trailingZeroes + units
```
Comment (guggero) [inline btcutil/amount.go:118]: nit: lines too long.

File: btcutil/amount.go [hunk 3]
```diff
@@ -101,11 +102,23 @@ func (a Amount) ToBTC() float64 {
...
 func (a Amount) Format(u AmountUnit) string {
 	units := " " + u.String()
-	return strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64) + units
+	formatted := strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64)
+
+	// when formatting full BTC, add trailing zeroes for numbers with decimal
+	// point to ease reading of sat amount
+	if u == AmountBTC {
+		decimalPoint := "."
+		before, after, containsDecimalPoint := strings.Cut(formatted, decimalPoint)
```
Comment (guggero) [inline btcutil/amount.go:115]: `strings.Cut` was only added in Go `1.18`, but our `go.mod` still says `Go 1.16`.
Also, why not just:
```
		if strings.Contains(formatted, ".") {
			return fmt.Sprintf("%.8f%s", a.ToUnit(u), units)
		}
```

**Review by guggero — COMMENTED** [review]
Thanks for the change, this has bugged me before as well.
Can be simplified quite a bit IMO.

**Comment by C-Otto** [comment]
Thanks, that's Go code written by a Java programmer :) I'll tidy it up!

**Review by guggero — APPROVED** [review]
Thanks, LGTM :tada:

File: btcutil/amount_test.go [hunk 4]
```diff
@@ -136,8 +136,22 @@ func TestAmountUnitConversions(t *testing.T) {
...
+			unit:      AmountBTC,
+			converted: 0.00001,
+			s:         "0.00001000 BTC",
+		},
+		{
+			name:      "a single satoshi as BTC",
+			amount:    1,
+			unit:      AmountBTC,
+			converted: 0.00000001,
+			s:         "0.00000001 BTC",
```
Comment (halseth) [inline btcutil/amount_test.go:154]: add a testcase for BTC without decimals (like 10.0 BTC)

**Review by halseth — COMMENTED** [review]
Code change ack, but could we accidentally break any code parsing these strings?

File: btcutil/amount_test.go (hunk 4 above)
Comment (C-Otto) [reply btcutil/amount_test.go:154]: Added

**Comment by C-Otto** [comment]
Yes, code parsing this might be affected. We're not changing the amount and 1.5 = 1.50, so it shouldn't cause massive headaches. It's a risk, though, and I'm not in a position to decide for or against this.

**Review by guggero — APPROVED** [review]

**Comment by guggero** [comment]
cc @Roasbeef

**Review by Roasbeef — APPROVED** [review]
LGTM 📸

**Pull request merged**